src/main/java/net/neoforged/neoforge/energy/EnergyStorage.java

```diff
@@ -23,6 +26,9 @@ public class EnergyStorage implements IEnergyStorage, INBTSerializable<Tag> {
     protected int maxReceive;
     protected int maxExtract;
 
+    public final Codec<EnergyStorage> CODEC;
```

Comment:
This is a weird pattern I think. Not a fan. Maybe we should have a static method that returns the codec, and people could use the codec to construct a default instance?